src/Microsoft.ML.Data/Transforms/KeyToVectorTransform.cs

```diff
@@ -206,393 +187,605 @@ public override void Save(ModelSaveContext ctx)
             ctx.SetVersionInfo(GetVersionInfo());
 
             // *** Binary format ***
-            // int: sizeof(Float)
             // <base>
             // for each added column
             //   byte: bag as 0/1
-            ctx.Writer.Write(sizeof(Float));
-            SaveBase(ctx);
+            // for each added column
+            //  int: keyCount
+            //  int: valueCount
+            SaveColumns(ctx);
+
+            Host.Assert(_bags.Length == ColumnPairs.Length);
+            for (int i = 0; i < _bags.Length; i++)
+                ctx.Writer.WriteBoolByte(_bags[i]);
+            Host.Assert(_valueCounts.Length == ColumnPairs.Length);
+            Host.Assert(_sizes.Length == ColumnPairs.Length);
 
-            Host.Assert(_bag.Length == Infos.Length);
-            for (int i = 0; i < _bag.Length; i++)
-                ctx.Writer.WriteBoolByte(_bag[i]);
+            for (int i = 0; i < ColumnPairs.Length; i++)
+            {
+                ctx.Writer.Write(_sizes[i]);
+                ctx.Writer.Write(_valueCounts[i]);
+            }
         }
 
-        public override bool CanSavePfa => true;
-        public override bool CanSaveOnnx => true;
+        // Factory method for SignatureLoadModel.
```

Comment:
>// Factor [](start = 8, length = 9)

Convert it to XML doc, since it is a public method (and if there is anything more to add)